tests/integration/callbacks/test_missing_inputs.py

```diff
@@ -0,0 +1,484 @@
+            html.Div("output2 init", id="out2"),
+        ]
+    )
+
+    @app.callback(Output("content", "children"), [Input("btn", "n_clicks")])
+    def content(n):
+        out = [html.Div("item {}".format(i), id={"i": i}) for i in range(n or 0)]
+        return out or "content init"
+
+    @app.callback(Output("out1", "children"), [Input({"i": ALL}, "children")])
+    def out1(items):
+        return ", ".join(items) or "no items"
+
+    @app.callback(Output("out2", "children"), [Input("out1", "children")])
+    def out2(out1):
+        return out1 + " - 2"
+
+    dash_duo.start_server(app)
+    wait_for_queue(dash_duo)
+
+    dash_duo.wait_for_text_to_equal("#out2", "no items - 2")
+
+    assert dash_duo.find_element("#out1").text == "no items"
+    assert dash_duo.find_element("#content").text == "content init"
+
+    dash_duo.find_element("#btn").click()
+    dash_duo.wait_for_text_to_equal("#out1", "item 0")
+    dash_duo.wait_for_text_to_equal("#out2", "item 0 - 2")
+
+    dash_duo.find_element("#btn").click()
+    dash_duo.wait_for_text_to_equal("#out1", "item 0, item 1")
+    dash_duo.wait_for_text_to_equal("#out2", "item 0, item 1 - 2")
+
+    assert not dash_duo.get_logs()
+
+
+def test_cbmi006_all_multi_wildcards_no_outputs(dash_duo):
+    # if all the inputs are multi wildcards, but there's NO output,
+    # we DO NOT fire the callback
+    app = dash.Dash(__name__, suppress_callback_exceptions=True)
+    app.layout = html.Div(
+        [
+            html.Div("Title", id="title"),
+            html.Button("click", id="btn"),
+            html.Div(id="content"),
+            html.Div("output2 init", id="out2"),
+        ]
+    )
+
+    @app.callback(Output("content", "children"), [Input("btn", "n_clicks")])
+    def content(n):
+        out = [html.Div("item {}".format(i), id={"i": i}) for i in range(n or 0)]
+        return (out + [html.Div("output1 init", id="out1")]) if out else "content init"
+
+    @app.callback(Output("out1", "children"), [Input({"i": ALL}, "children")])
+    def out1(items):
+        return ", ".join(items) or "no items"
+
+    @app.callback(Output("out2", "children"), [Input("out1", "children")])
+    def out2(out1):
+        return out1 + " - 2"
+
+    dash_duo.start_server(app)
+    wait_for_queue(dash_duo)
+
+    dash_duo.wait_for_text_to_equal("#out2", "output2 init")
+
+    assert dash_duo.find_element("#content").text == "content init"
+
+    dash_duo.find_element("#btn").click()
+    dash_duo.wait_for_text_to_equal("#out1", "item 0")
+    dash_duo.wait_for_text_to_equal("#out2", "item 0 - 2")
+
+    dash_duo.find_element("#btn").click()
+    dash_duo.wait_for_text_to_equal("#out1", "item 0, item 1")
+    dash_duo.wait_for_text_to_equal("#out2", "item 0, item 1 - 2")
+
+    assert not dash_duo.get_logs()
+
+
+def test_cbmi007_all_multi_wildcards_some_outputs(dash_duo):
+    # same as above (cbmi006) but multi-output, some outputs present some missing.
+    # Again we DO NOT fire the callback
```

Comment:
This is a case I didn't think of until writing related tests, but I think it makes sense, because if there's any missing output and no inputs, that output will probably show up when the inputs do.

This is basically the same as test `cbmi004` just for the case when all the inputs have multi-value wildcards so we do _try_ to fire the callback, and we need to bail out at a later step when we collect all the outputs.